### PR TITLE
Bump go deps

### PR DESCRIPTION
**Pull request by case-fastly**

### CODE REVIEW

**Review by ydnar — APPROVED** [review]

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch: # Enables the web UI manual trigger
+
+jobs:         
+  test:
+    name: Test
+    runs-on: ubuntu-22.04
```
Comment (ydnar) [inline .github/workflows/go.yaml:13]: Can this be ubuntu-latest?

File: .github/workflows/go.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    timeout-minutes: 15
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4.2.0
+        with:
+          submodules: recursive
+
+      - name: Set up Go
+        uses: actions/setup-go@v5.0.2
```
Comment (ydnar) [inline .github/workflows/go.yaml:22]: @5 (so it uses latest patch version)

File: .github/workflows/go.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  workflow_dispatch: # Enables the web UI manual trigger
+
+jobs:         
+  test:
+    name: Test
+    runs-on: ubuntu-22.04
+    timeout-minutes: 15
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4.2.0
```
Comment (ydnar) [inline .github/workflows/go.yaml:17]: @4 (so it uses latest patch version)

File: .github/workflows/go.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v5.0.2
+        with:
+          go-version: ^1
+
+      - name: Vet Go code
+        run: go vet ./...
+
+      - name: govulncheck
+        uses: golang/govulncheck-action@v1.0.3
```
Comment (ydnar) [inline .github/workflows/go.yaml:30]: Same here

**Review by ydnar — CHANGES_REQUESTED** [review]

**Comment by case-fastly** [comment]
I see you have high confidence in non-specific version numbers. 😂

(I prefer specific versions, so that we can intentionally bump them, and know with more certainty if anything breaks or changes as a result.)

**Comment by case-fastly** [comment]
Discussed these comments with @ydnar IRL, he's game for these to stay as precise versions.

**Comment by ydnar** [comment]
I lost approval rights on this repo

**Pull request merged**